src/api/app/models/user.rb

```diff
@@ -903,6 +900,16 @@ def update_notifications(params)
 
   private
 
+  def involved_packages_ids
+    user_maintained_packages.where.not(project_id: involved_projects_ids).pluck(:id) +
+        group_maintained_packages.where.not(project_id: involved_projects_ids).pluck(:id)
```

Comment:
They're both relationships and they're structurally incompatible, so e.g.

``
user_maintained_packages.or(group_maintained_packages)
``
does not work:
``
Relation passed to #or must be structurally compatible. Incompatible values: [:joins, :references]
``
(They joins are different: User -> Relationship -> Package vs. User -> Groups -> Relationships -> Package)

I also don't really like this ... :cry: 

Same is also the case for involved_projects